docs/aws/audit/sgaudit/rules/ec2_instances_with_icmp_exist.mdx

```diff
@@ -23,6 +23,222 @@ NIST
+   ```sh
+   aws ec2 describe-security-groups --query 'SecurityGroups[*].[GroupId,GroupName]' --output table
+   ```
+
+2. **Check Existing Rules:**
+   Check the existing inbound rules for the identified security group to see if there are any open ICMP ports:
+   ```sh
+   aws ec2 describe-security-groups --group-ids <security-group-id> --query 'SecurityGroups[*].IpPermissions'
+   ```
+
+3. **Revoke Open ICMP Rules:**
+   If you find any open ICMP rules, revoke them using the following command. Replace `<security-group-id>` with your security group ID and adjust the `--protocol`, `--port`, and `--cidr` parameters as needed:
+   ```sh
+   aws ec2 revoke-security-group-ingress --group-id <security-group-id> --protocol icmp --port all --cidr 0.0.0.0/0
+   ```
+
+4. **Add Specific ICMP Rules (if needed):**
+   If you need to allow specific ICMP types (e.g., Echo Request), you can add more restrictive rules. For example, to allow only ICMP Echo Request (type 8, code 0) from a specific IP range:
+   ```sh
+   aws ec2 authorize-security-group-ingress --group-id <security-group-id> --protocol icmp --port -1 --cidr <specific-cidr> --icmp-type-code 8:0
+   ```
+
+By following these steps, you can ensure that your EC2 instances do not have open ICMP ports in their security groups, thereby enhancing your security posture.
+</Accordion>
+
+<Accordion title='Using Python'>
+To prevent EC2 instances from having open ICMP ports in their security groups using Python scripts, you can use the AWS SDK for Python (Boto3). Here are the steps:
+
+1. **Install Boto3**:
+   Ensure you have Boto3 installed. If not, you can install it using pip:
+   ```bash
+   pip install boto3
+   ```
+
+2. **Initialize Boto3 Client**:
+   Initialize the Boto3 client for EC2.
+
+3. **Describe Security Groups**:
+   Retrieve the security groups and their rules.
+
+4. **Check and Update Security Groups**:
+   Check for open ICMP ports and update the security groups to remove or restrict them.
+
+Here is a Python script to achieve this:
+
+```python
+import boto3
+
+# Initialize a session using Amazon EC2
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
+    region_name='YOUR_REGION'
+)
+
+ec2 = session.client('ec2')
+
+# Describe all security groups
+response = ec2.describe_security_groups()
+
+for sg in response['SecurityGroups']:
+    sg_id = sg['GroupId']
+    ip_permissions = sg['IpPermissions']
+    
+    for permission in ip_permissions:
+        # Check if the protocol is ICMP (protocol number 1)
+        if permission['IpProtocol'] == 'icmp':
+            print(f"Security Group {sg_id} has open ICMP ports.")
+            
+            # Revoke the ICMP permission
+            ec2.revoke_security_group_ingress(
+                GroupId=sg_id,
+                IpPermissions=[
+                    {
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>